tests/test_client.py

```diff
@@ -357,17 +347,16 @@ def test_save_to_well_known_file_service_account(self):
         self.assertEqual('ABCDEF', d['private_key_id'])
         os.remove(temp_credential_file)
 
-    def test_save_well_known_file_with_non_existent_config_dir(self):
+    @mock.patch('os.path.isdir', return_value=False)
+    def test_save_well_known_file_with_non_existent_config_dir(self,
```

Comment:
I don't understand the change on this line. Mustn't `self` be the exactly one parameter of a test method?